lib/Raven/ReprSerializer.php

```diff
@@ -29,7 +29,11 @@ protected function serializeValue($value)
         } elseif (is_integer($value) || is_float($value)) {
             return (string) $value;
         } elseif (is_object($value) || gettype($value) == 'object') {
-            return 'Object '.get_class($value);
+            $name = 'Object '.get_class($value);
+            if (method_exists($value, '__toString')) {
```

Comment:
i think the concern here would be when objects return gigantic values for strings (also you'd need to gracefully handle errors with ``$value->__toString()`` since its user code and could easily error)

Reply:
I could limit it to 1000 characters with substr, though that will not help if `__toString` exceeds the memory limit. Ignore that?

Regarding error handling I would do:

* for exceptions: wrap it in try-catch 
* for PHP errors firing the error handler: use custom handler, afterwards call `restore_error_handler`
* for PHP errors causing shutdown: this is not recoverable, so I would push the error to sentry, but `ReprSerializer` doesn't have access to a `Client`. Apart from tests, there are only two instantiations of `ReprSerializer`. The client could be passed in:
  * Stacktrace.php:27
  * Client.php:204